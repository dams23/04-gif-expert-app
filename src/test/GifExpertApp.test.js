import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GitExpertApp'

describe('pruebas sobre el componente GifExpertApp', () => {
    test('debe coincidir con el snap', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de mostrar una lista de categorias', () => {
        const categories = ['Star Wars', 'Han Solo'];
        const wrapper = shallow(<GifExpertApp
            defaultCategories={categories}
        />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});