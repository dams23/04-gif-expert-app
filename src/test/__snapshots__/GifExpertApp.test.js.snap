// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre el componente GifExpertApp debe coincidir con el snap 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`pruebas sobre el componente GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Star Wars"
      key="Star Wars"
    />
    <GifGrid
      category="Han Solo"
      key="Han Solo"
    />
  </ol>
</Fragment>
`;
